src/recordpage.cpp

```diff
@@ -249,3 +357,21 @@ void RecordPage::onOutputFileUrlChanged(const QUrl& fileUrl)
 {
     onOutputFileNameSelected(fileUrl.toLocalFile());
 }
+
+void RecordPage::updateProcesses()
+{
+    auto *watcher = new QFutureWatcher<ProcDataList>(this);
+    connect(watcher, SIGNAL(finished()),
+            this, SLOT(updateProcessesFinished()));
+    watcher->setFuture(QtConcurrent::run(processList, m_processModel->processes()));
+}
+
+void RecordPage::updateProcessesFinished()
+{
+    QFutureWatcher<ProcDataList> *watcher = dynamic_cast<QFutureWatcher<ProcDataList> *>(sender());
```

Comment:
couldn't you reuse the same watcher, instead of recreating it all the time? or is that not supported? I have not a lot of experience with that API